src/burnchains/burnchain.rs

```diff
@@ -1213,12 +1213,21 @@ impl Burnchain {
         );
 
         if let Some(target_block_height) = target_block_height_opt {
-            if target_block_height < end_block {
+            // target_block_height is used as a hint, but could also be completely off
+            // in certain situations. The current function is directly reading the
+            // headers and syncing with the bitcoin-node, and the interval of blocks
+            // to download computed here should be considered as our source of truth.
```

Comment:
Does the comment on line 1144 need to change?

Reply:
I don't think so. This function is still returning a burnchain tip, at least as high as target_block_height_opt if given.